docs/en/week08/08-3.md

```diff
@@ -0,0 +1,266 @@
+
+- First, the autoencoder takes in an input and maps it to a hidden state through an affine transformation $\boldsymbol{h} = f(\boldsymbol{W}_h \boldsymbol{x} + \boldsymbol{b}_h)$, where $f$ is an (element-wise) activation function. This is the **encoder** stage. Note that $\boldsymbol{h}$ is also called the **code**.
+- Next, $\hat{\boldsymbol{x}} = g(\boldsymbol{W}_x \boldsymbol{h} + \boldsymbol{b}_x)$, where $g$ is an activation function. This is the **decoder** stage.
+
+
+### Reconstruction costs
+
+When the input is real-valued we use mean squared error loss, 
+
+$$
+C(\boldsymbol{y},\boldsymbol{\tilde{y}}) = \Vert \boldsymbol{y} - \boldsymbol{\tilde{y}} \Vert^{2} =  \Vert{\boldsymbol{y} - \text{Dec}[\text{Enc}(\boldsymbol{y})]}\Vert^{2}
+$$
+
+When the input is categorical we use cross entropy loss,
+
+$$
+C(\boldsymbol{y},\boldsymbol{\tilde{y}}) = -\sum_{i = 1}^{n} [y_{i} \log(\tilde{y_{i}}) + (1 - y_{i})\log(1 - \tilde{y_{i}})]
+$$
+
+
+### Loss functional
+
+The loss functional is the average of the per sample loss.
+
+$$
+\mathcal{L}(F(\cdot), Y) = \frac{1}{m} \sum_{j = 1}^{m} \ell(F(\cdot),y^{(j)}) \in \mathbb{R} \\
+\ell_\text{energy}(F(\cdot),\boldsymbol{y}) = F(\boldsymbol{y})
+$$
+
+
+### Under-/over-complete hidden layer
+
+When the dimensionality of the hidden layer is less than the dimensionality of the input then it is under complete hidden layer. When the dimensionality of the hidden layer is greater than the dimensionality of the input then it is over complete. 
+
+The dimensionality of the hidden layer is larger than the input so that they can be linearly seperable. However, this can lead to a collapsed representation. Since by reconstructing the input, the model can copy all the features. In order to avoid this there are a few techniques such as denoising autoencoder, contractive autoencoder, variation autoencoder, etc. 
+
+
+## Denoising Autoencoder
+
+Denoising Autoencoder is a contrastive technique. *Fig. 3* shows the manifold of the denoising autoencoder and the intuition of how it works. 
+
+<center>
+<img src="{{site.baseurl}}/images/week08/08-3/DAE.png" style="background-color:#DCDCDC;" /><br>
+<b>Fig. 3</b>: Denoising Autoencoder
+</center>
+
+In this model, we corrupt the input data <span style="color:#80b1d3">$\boldsymbol{y}$</span>, and get the sample <span style="color:#fb8072">$\boldsymbol{\hat{y}}$</span> which has a high energy. The energy associated to $\boldsymbol{\tilde y}$ is squared euclidian distance from its original value. Then $\boldsymbol{\tilde y}$ is encoded back to the hidden variable $h$ and then passed back to the decoder producing $\boldsymbol{\tilde y}$ which should be close to the target <span style="color:#80b1d3">$\boldsymbol{y}$</span>. Hence irrespective of what noise is added to the original data, the autoencoder needs to learn to separate the noise and retrieve the original value of <span style="color:#80b1d3">$\boldsymbol{y}$</span>.
```

Comment:
input data y, in blue, because it's cold / low energy
y-hat, in red, because it's hot / high energy

Reply:
added